client/lib/signup/flow-controller.js

```diff
@@ -130,6 +130,45 @@ assign( SignupFlowController.prototype, {
 		}.bind( this ) );
 	},
 
+	/**
+	 * This assertion helper validates that there are dependencies for the step provided
+	 * by completed steps
+	 * @param step The step to validate
+	 */
+	_assertStepDependencies( step ) {
```

Comment:
I'm trying to understand why this function is needed, because we currently assert that all dependencies listed as provided in the config are provided in `_assertFlowProvidedRequiredDependencies`. Also, what do you think about using a more specific name here (`_assertStepProvidesDependencies`?).